translator/source_code/icone_translator.py

```diff
@@ -46,14 +57,17 @@ def wzdx_creator(messages, info):
     ids['road_event_id'] = road_event_id
 
     wzd['features'] = []
-    for incident in messages['incidents']['incident']:
-        # Parse Incident to WZDx Feature
 
+    if not messages.get('incidents', {}).get('incident'):
+        return None
 
-        feature = parse_incident(incident)
+    for incident in messages['incidents']['incident']:
+        # Parse Incident to WZDx Feature    
+        feature = parse_incident(incident, callback_function=unsupported_message_callback)
         if feature:
             wzd['features'].append(feature)
-
+    if not wzd['features']:
+        return None
```

Comment:
Is this the correct logic here? We're saying that if we find any incidents with invalid features we don't generate any WZDx messages. So we might have 50 incidents and if a single one fails the full ingest doesn't succeed. Is this right?

Reply:
wzd['features'] is the compiled list of features, checked after the for look which populates it. If it is completely empty (not a single incident), then it will return None instead of returning a complete WZDx object with an empty list of features

Reply:
Ohh I got it, that makes sense then!